ci-operator/step-registry/baremetal/lab/agent/add-day2-nodes/baremetal-lab-agent-add-day2-nodes-commands.sh

```diff
@@ -0,0 +1,220 @@
+
+curl https://github.com/openshift/installer/blob/master/docs/user/agent/add-node/node-joiner.sh --output "${INSTALL_DIR}/node-joiner.sh"
+
+cat > "${INSTALL_DIR}/nodes-config.yaml" <<EOF
+hosts: []
+EOF
+
+# shellcheck disable=SC2154
+for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
+  # shellcheck disable=SC1090
+  . <(echo "$bmhost" | yq e 'to_entries | .[] | (.key + "=\"" + .value + "\"")')
+  if [[ "${name}" == *-a-* ]] && [ "${ADDITIONAL_WORKERS_DAY2}" == "true" ]; then
+
+  ADAPTED_YAML="
+  hostname: ${name}
+  interfaces:
+  - macAddress: ${mac}
+    name: ${baremetal_iface}
+  networkConfig:
+    interfaces:
+    - name: ${baremetal_iface}
+      type: ethernet
+      state: up
+      ipv4:
+        enabled: ${ipv4_enabled}
+        dhcp: false
+        address:
+            - ip: ${ip}
+              prefix-length: ${INTERNAL_NET_CIDR##*/}
+      ipv6:
+        enabled: ${ipv6_enabled}
+"
+
+  # split the ipi_disabled_ifaces semi-comma separated list into an array
+  IFS=';' read -r -a ipi_disabled_ifaces <<< "${ipi_disabled_ifaces}"
+  for iface in "${ipi_disabled_ifaces[@]}"; do
+    # Take care of the indentation when adding the disabled interfaces to the above yaml
+    ADAPTED_YAML+="
+    - name: ${iface}
+      type: ethernet
+      state: up
+      ipv4:
+        enabled: false
+        dhcp: false
+      ipv6:
+        enabled: false
+        dhcp: false
+    "
+  done
+
+  # Take care of the indentation when adding the dns and routes to the above yaml
+  ADAPTED_YAML+="
+    dns-resolver:
+          config:
+            server:
+              - ${INTERNAL_NET_IP}
+    routes:
+      config:
+        - destination: 0.0.0.0/0
+          next-hop-address: ${INTERNAL_NET_IP}
+          next-hop-interface: ${baremetal_iface}
+  "
+  # Patch the nodes-config.yaml by adding the given host to the hosts list in the platform.baremetal stanza
+  yq --inplace eval-all 'select(fileIndex == 0).hosts += select(fileIndex == 1) | select(fileIndex == 0)' \
+    "$INSTALL_DIR/nodes-config.yaml" - <<< "$ADAPTED_YAML"
+  fi
+done
+
+cp "${INSTALL_DIR}/nodes-config.yaml" "${INSTALL_DIR}/"
+
+export KUBECONFIG="$SHARED_DIR/kubeconfig"
+
+chmod +x "${INSTALL_DIR}/node-joiner.sh"
+sh "${INSTALL_DIR}/node-joiner.sh"
+
+# Patching the cluster_name again as the one set in the ipi-conf ref is using the ${UNIQUE_HASH} variable, and
+# we might exceed the maximum length for some entity names we define
+# (e.g., hostname, NFV-related interface names, etc...)
+CLUSTER_NAME=$(<"${SHARED_DIR}/cluster_name")
+
+
+gnu_arch=$(echo "$architecture" | sed 's/arm64/aarch64/;s/amd64/x86_64/;')
+case "${BOOT_MODE}" in
+"iso")
+  ### Create ISO image
+  #echo -e "\nCreating image..."
+  #oinst agent create image
+  ### Copy the image to the auxiliary host
+  echo -e "\nCopying the day2 node ISO image into the bastion host..."
+  scp "${SSHOPTS[@]}" "${INSTALL_DIR}/node.x86_64.iso" "root@${AUX_HOST}:/opt/html/${CLUSTER_NAME}.node.iso"
```

Comment:
note that this should consider other arch, and heterogeneous clusters too (day2 nodes of a different architecture than the control plane).

cc @zniu1011

Reply:
Multiarch case is not supported in the initial version of `node-joiner.sh`, intend to support in the onwards versions. And we are planning to release `oc` command line instead of `node-joiner.sh` like:  `oc adm add-node-images `in 4.17, we can consider to use agent day2 install way for the heterogeneous cases on the Agent side then.

Reply:
@aleskandro yes, we know about agent day2. We haven't prioritized multi-arch there yet. My best guess would be 4.18?

Reply:
Also @zniu1011 @bmanzari, would this work on single-arch arm64 at least (which is GAed already)?

Reply:
@aleskandro , arm64 is not supported yet.

Reply:
> @aleskandro yes, we know about agent day2. We haven't prioritized multi-arch there yet. My best guess would be 4.18?

Do we have a story/epic for this already? Do we also need  one for arm64 at this point?

Reply:
> @aleskandro yes, we know about agent day2. We haven't prioritized multi-arch there yet. My best guess would be 4.18?

Do we have a story/epic for this already? Do we also need one for arm64 at this point?